app/services/company.js

```diff
@@ -61,35 +109,43 @@ class CompanyService extends CrudService {
   }
 
   async update(id, data) {
-    if (data.id !== undefined) {
-      error.throwValidationError('You can not change the id.');
-    }
-
-    if (data.name !== undefined) {
-      error.throwValidationError('You can not change the company name.');
-    }
+    await companyValidator.validateUpdate(data);
 
     const { business, ...newData } = data;
     if (business !== undefined) {
-      const fieldOfActivity = await fieldOfActivityRepository.getByPropsNonParanoid({ name: business });
+      const fieldOfActivity = await fieldOfActivityRepository.getByPropsNonParanoid({
+        name: business,
+      });
       if (!fieldOfActivity) {
         error.throwNotFoundError('Field of activity not found.');
       }
       newData.fieldOfActivityId = fieldOfActivity.id;
     }
 
-    await this.get(id);
+    const companyExists = await this.repository.get(id);
```

Comment:
1) `companyExists` makes me think this is a boolean value, but in fact it is a company. Please rename it.
2) this check should be in the validator.